ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -269,6 +286,25 @@ Output getOutput(Env env) {
               }
               throw (RuntimeException) t;
             }).thenCompose(v -> v);
+        // Send messages not matched by STREAMING_DOCTYPES directly to fileOutput
+        final Function<PubsubMessage, CompletableFuture<Void>> mixedOutput;
+        if (env.containsKey(STREAMING_DOCTYPES)) {
+          Pattern streamingDoctypes = env.getPattern(STREAMING_DOCTYPES);
+          PubsubMessageToTemplatedString doctypeTemplate = PubsubMessageToTemplatedString
+              .of("${document_namespace}/${document_type}");
+          mixedOutput = message -> {
+            try {
+              if (streamingDoctypes.matcher(doctypeTemplate.apply(message)).matches()) {
```

Comment:
Anywhere we're evaluating a regex on a per-message basis, I think we should be caching the result. Can we introduce a guava cache here?